buffer.go

```diff
@@ -256,7 +256,7 @@ func (buf *Buffer) WriteRowGroup(rowGroup RowGroup) (int64, error) {
 //
 // The buffer and the returned reader share memory. Mutating the buffer
 // concurrently to reading rows may result in non-deterministic behavior.
-func (buf *Buffer) Rows() Rows { return &rowGroupRows{rowGroup: buf} }
+func (buf *Buffer) Rows() Rows { return newRowGroupRows(buf, PageReadModeAsync) }
```

Comment:
Should we use the sync mode here since the buffer holds values in memory?

Reply:
I agree with your thoughts below about defaulting to Sync because it creates the least surprises. In all places where we create `newRowGroupRows()` should we default to `PageReadModeSync`?

Reply:
changed to sync